.github/workflows/ci.yml

```diff
@@ -94,3 +92,16 @@ jobs:
           stack install
           stack exec -- ihaskell install --stack
           test/acceptance.nbconvert.sh stack exec -- jupyter nbconvert
+      - name: Check Dockerfile GHC version matches
```

Comment:
I have to say it's pretty funny that the solution you've ended up with still has the same flaw w.r.t multiple sources of truth that our current setup does. We could add these lines alone and get 90% of the benefit of this PR.

Reply:
Yeah, that's true, although I disagree about the 90% benefit. I would say the biggest improvement this PR makes is bringing the image size down from 6 GB to 1.5 GB.

Reply:
And actually, the original multiple-sources-of-truth dealt with the `~/.stack/global-project/stack.yaml` resolver and the in-repo `stack.yaml` resolver. That is now one source of truth here: https://github.com/gibiansky/IHaskell/pull/1252/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R91

The `GHC_VERSION` in the `Dockerfile` is just the GHC preloaded into the Docker image. So actually, if `stack.yaml` resolver used a different version of GHC, it would just download another GHC into the build image, but the runtime image's global stack resolver would still match the in-repo stack resolver.

Reply:
I don't really care about that, since an IHaskell Docker image built with Nix is also quite small: https://vaibhavsagar.com/blog/2019/08/11/ihaskell-nix-docker/.

Reply:
Actually the Nix-built size comparison isn't fair, I just imported the archive and the resulting image is nearly 4GB so your Dockerfile is in fact the one that results in an image that is more than twice as small.

Reply:
If by chance you want to see for yourself, the Nix expression is here: https://gist.github.com/vaibhavsagar/1cf8ca927a8ee872d43619c4ac2d2665.

Reply:
> Also, you've moved the problem at the expense of additional complexity in the Dockerfile and the IHaskell source code itself that I would have to maintain and update in the presence of changes, so it's arguable whether this is an improvement.

Moving the problem is separate from the additional complexity I added to the Dockerfile/source code. I could've solved this problem by making a PR with just
```diff
@@ line 4 @@
-ARG RESOLVER=lts-17.4

@@ line 49 @@
 RUN stack build && stack install
 RUN mkdir -p ${HOME}/.stack/global-project && \
+    RESOLVER=$(sed -n 's/resolver: \(.*\)/\1/p' stack.yaml) && \
     echo "packages: []\nresolver: ${RESOLVER}\n" > ${HOME}/.stack/global-project/stack.yaml
```
All the complexity in this PR is due to me bringing Dockerfile best practices to the Dockerfile to produce a smaller final image using the `haskell` base docker image (which allowed us to avoid manually installing `stack`)

> It's never explicitly been our focus to provide a production-quality Docker image.

That's fair, but I would also ask why it's not. As a user, it's a bit difficult to quickly install + get set up with IHaskell if I need to install a bunch of dependencies and build it from scratch with Stack. If there were pre-built binaries that one could download, that would help, but since IHaskell links a bunch of C libraries (e.g. cairo), one would still need to install dependencies, and I personally try to avoid globally installing extra dependencies.

I might be open to taking ownership of maintaining the `Dockerfile` in this repo, if that's your primary concern.